collectors/ebpf.plugin/ebpf_dcstat.c

```diff
@@ -0,0 +1,604 @@
+ *
+ *  COMMON FUNCTIONS
+ *
+ *****************************************************************/
+
+/**
+ * Update publish
+ *
+ * Update publish values before to write dimension.
+ *
+ * @param out           strcuture that will receive data.
+ * @param cache_access  number of access to directory cache.
+ * @param not_found     number of files not found on the file system
+ */
+void dcstat_update_publish(netdata_publish_dcstat_t *out, uint64_t cache_access, uint64_t not_found)
+{
+    calculated_number successful_access = (calculated_number) (((long long)cache_access) - ((long long)not_found));
+    calculated_number ratio = (cache_access) ? successful_access/(calculated_number)cache_access : 0;
+
+    out->ratio = (long long )(ratio*100);
+}
+
+/*****************************************************************
+ *
+ *  FUNCTIONS TO CLOSE THE THREAD
+ *
+ *****************************************************************/
+
+/**
+ * Clean PID structures
+ *
+ * Clean the allocated structures.
+ */
+void clean_dcstat_pid_structures() {
+    struct pid_stat *pids = root_of_pids;
+    while (pids) {
+        freez(dcstat_pid[pids->pid]);
+
+        pids = pids->next;
+    }
+}
+
+/**
+ *  Clean names
+ *
+ *  Clean the optional names allocated during startup.
+ */
+void ebpf_dcstat_clean_names()
+{
+    size_t i = 0;
+    while (dc_optional_name[i].program_name) {
+        freez(dc_optional_name[i].optional);
+        i++;
+    }
+}
+
+/**
+ * Clean up the main thread.
+ *
+ * @param ptr thread data.
+ */
+static void ebpf_dcstat_cleanup(void *ptr)
+{
+    ebpf_module_t *em = (ebpf_module_t *)ptr;
+    if (!em->enabled)
+        return;
+
+    heartbeat_t hb;
+    heartbeat_init(&hb);
+    uint32_t tick = 2 * USEC_PER_MS;
+    while (!read_thread_closed) {
+        usec_t dt = heartbeat_next(&hb, tick);
+        UNUSED(dt);
+    }
+
+    freez(dcstat_vector);
+
+    ebpf_cleanup_publish_syscall(dcstat_counter_publish_aggregated);
+
+    ebpf_dcstat_clean_names();
+
+    struct bpf_program *prog;
+    size_t i = 0 ;
+    bpf_object__for_each_program(prog, objects) {
+        bpf_link__destroy(probe_links[i]);
+        i++;
+    }
+    bpf_object__close(objects);
+}
+
+/*****************************************************************
+ *
+ *  APPS
+ *
+ *****************************************************************/
+
+/**
+ * Create apps charts
+ *
+ * Call ebpf_create_chart to create the charts on apps submenu.
+ *
+ * @param em a pointer to the structure with the default values.
+ */
+void ebpf_dcstat_create_apps_charts(struct ebpf_module *em, void *ptr)
+{
+    UNUSED(em);
+    struct target *root = ptr;
+    ebpf_create_charts_on_apps(NETDATA_DC_HIT_CHART,
+                               "Percentage of files listed inside directory cache",
+                               EBPF_COMMON_DIMENSION_PERCENTAGE,
+                               NETDATA_APPS_DCSTAT_GROUP,
+                               NETDATA_EBPF_CHART_TYPE_STACKED,
```

Comment:
```suggestion
                               NETDATA_EBPF_CHART_TYPE_LINE,
```
Please do the same for `apps.cachestat_ratio` (in a separate PR?).

Reply:
Yes, I will bring a fix for this case in a separated PR.